source/_posts/2016-01-25-upgrading-to-ruby-on-rails-5-0.markdown

```diff
@@ -0,0 +1,226 @@
+
+From the upgrade guide (still very inadequate), we learn that some of the changes in Rails 5.0 are:
+
+- Requiring that we use at least Ruby 2.2.2 or newer
+- The models inherit from ApplicationRecord 
+- The jobs inherit from ApplicationJob (although inheriting from Application* isn’t required)
+- Using throw(:abort) instead of returning false for callback halting.
+
+After familiarizing ourselves a bit (in the article we skip the topic of CHANGELOGs), we can move on to updating the application. We’ll show you how we ourselves update a small application that we’re developing, obviously keeping in mind that every application is going to present its own challenges and peculiarities.
+
+
+## First naive attempt to upgrade
+
+Given that our application already worked with Ruby 2.2.2, we didn’t have to do anything in this respect.
+
+The first step we took was to modify the Gemfile by changing the Rails version from 4.2.5 to 5.0.0.beta1. The Gemfile needs to have a line with `gem 'rails', '5.0.0.beta1'`.
+
+The following step is to execute `bundle update rails` and ...
+
+```
+    devise was resolved to 3.5.2, which depends on
+      railties (< 5, >= 3.2.6)
+```
+
+It was a very likely possibility that some gems were still not compatible with Rails 5. In the case of devise, the master version does indeed support Rails 5 and we therefore change the devise line in the `Gemfile` to `gem 'devise', github: 'plataformatec/devise'`.
+
+We again execute `bundle update rails` and ...
+
+```
+    minitest-rails was resolved to 2.2.0, which depends on
+      railties (~> 4.1)
+```
+
+For this gem, there’s no version that will fix the problem. Still, the first thing I asked myself was, what things were we using from that gem. The gem sets up the minitest spec DSL and adds some generators, but we weren’t using the DSL, nor the generators, so we simply erase this Gemfile entry. 
+
+We execute `bundle update rails` again ...
+
+```
+    minitest-rails-capybara was resolved to 2.1.1, which depends on
+      minitest-rails (~> 2.1) was resolved to 2.2.0, which depends on
+        railties (~> 4.1)
+```
+
+We are using minitest-rails-capybara, which depends on minitest-rails. We just erased the Gemfile, but owing to this dependency, minitest-rails continues trying to install. We follow the same procedure from the previous case, asking ourselves, “What do we need minitest-rails-capybara for?” The reality is that we don’t need it either, we’ll simply use minitest-capybara, which is one of its dependencies.
```

Comment:
`We are using minitest-rails-capybara, which depends on minitest-rails. We just erased the Gemfile, ` => `We are using minitest-rails-capybara, that depends on minitest-rails which we've just removed from the Gemfile, `